coa/pkg/apis/v1alpha2/providers/states/memorystate/memorystate.go

```diff
@@ -348,3 +341,39 @@ func (a *MemoryStateProvider) ReturnDeepCopy(s states.StateEntry) (states.StateE
 	}
 	return ret, nil
 }
+
+func (a *MemoryStateProvider) BumpGeneration(ctx context.Context, new map[string]interface{}, old map[string]interface{}) {
```

Comment:
Originally ETag is designed as the generic etag, and K8s generation is used as etag for K8s. Memory state provider maintains a simple ETag that is simply incremented every updates. Maybe we should use this auto-bumped generation as etag instead of the auto-incremented etag (see line 107).